sklearn/manifold/tests/test_locally_linear.py

```diff
@@ -136,6 +136,18 @@ def test_singular_matrix():
                   M, 2, 1, method='standard', eigen_solver='arpack')
 
 
+# Test the error raised if an neighbor object with an invalid metric is passed
+def test_invalid_metric():
+    rand = np.random.RandomState(0)
+    X = rand.randint(0, 100, size=(20, 3))
+    NN = neighbors.NearestNeighbors(2, metric='manhattan').fit(X)
+
+    msg = "metric must be euclidean or equivalent"
+    assert_raise_message(ValueError, msg, manifold.locally_linear_embedding,
+                         NN, 2, 1, method='standard', eigen_solver='arpack')
```

Comment:
I don't think method and eigen_solver are needed, right?

I would use keyword parameters, i.e. n_neighbors=2, n_components=1 because it's more readable (I don't think people know the signature of locally_linear_embedding by heart).

Reply:
Regarding the first, fixed.

Regarding the second, kept just so it's uniform with the previous test (but can change both)